src/__tests__/resolver.spec.ts

```diff
@@ -2070,6 +2070,64 @@ describe('resolver', () => {
     });
   });
 
+  describe('print tree', () => {
+    test('should handle local refs', async () => {
+      const data = {
+        title: 'Example',
+        type: 'object',
+        definitions: {
+          bear: {
+            type: 'object',
+            properties: {
+              type: {
+                type: 'string',
+              },
+              diet: {
+                type: 'string',
+              },
+              age: {
+                type: 'number',
+              },
+            },
+            required: ['type', 'diet', 'age'],
+          },
+        },
+        description: 'Bears are awesome',
+        properties: {
+          id: {
+            type: 'string',
+          },
+          bear: {
+            $ref: '#/definitions/bear',
+          },
+        },
+      };
+
+      const resolver = new Resolver();
+      const { graph } = await resolver.resolve(data);
+
+      expect(graph.serialize('root')).toMatchSnapshot();
+    });
+
+    // ./a#/foo -> ./b#bar -> ./a#/xxx -> ./c -> ./b#/zzz
+    test('should resolve http relative paths + back pointing uri refs', async () => {
```

Comment:
Would be nice to have a test case covering circular deps.

Reply:
Agreed will add one